### PR TITLE
[Compose] - 2582 - Implement search channels by name and participants

### PR DESCRIPTION
**Pull request by bychkovdmitry**
https://github.com/GetStream/stream-chat-android/issues/2582

### 🎯 Goal

Implement channels search by either channel name (for named channels) or member name (for distinct channels).

### 🛠 Implementation details

There is no easy way to check if the channel is distinct. As a workaround, I treat distinct channels as channels without name (which is not always the case).

```json
{
   "$or":[
      {
         "$and":[
            {
               "member.user.name":{
                  "$autocomplete":"Leandro"
               }
            },
            {
               "name":{
                  "$exists":false
               }
            }
         ]
      },
      {
         "name":{
            "$autocomplete":"Leandro"
         }
      }
   ]
}
```

### 🎨 UI Changes

https://user-images.githubusercontent.com/9600921/140742358-aeb94efa-6f32-46ff-b585-4d9fd21b1b09.mov

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [x] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS (task created))
- [x] Reviewers added

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -77,23 +77,23 @@
...
 - Improved the set of customization options for the `DefaultChannelItem`
 - Updated documentation for Channels set of features
+- Now it is possible to search for distinct channels by member names using `ChannelListViewModel`.
 
 ### ✅ Added
 - Added a new parameter to the `AttachmentFactory` called `previewContent` that represents attachments within the MessageInput
 - Added the `leadingContent`, `detailsContent`, `trailingContent` and `divider` Slot APIs for the `DefaultChannelItem`
+- Added `StreamDimens` option to the `ChatTheme`, to allow for dimension customization across the app.
+- Added localization support for the components related the channel list.
+- Added the `emptySearchContent` parameter to `ChannelList` component that allows to customize the empty placeholder, when there are no channels matching the search query.
```
Comment (filbabic) [inline CHANGELOG.md:94]: I'm wondering if all this is already merged to develop? Feels like something that shouldn't show up in this git diff

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/channel/ChannelListViewModel.kt [hunk 2]
```diff
@@ -97,25 +97,49 @@ public class ChannelListViewModel(
...
+     *
+     * If the [searchQuery] is empty, then returns the original [filter] provided by the user.
+     * Otherwise, returns a wrapped [filter] that also checks that the channel name match the
+     * [searchQuery].
+     *
+     * @param filter The filter that was passed by the user.
+     * @param searchQuery The search query used to filter the channels.
+     * @return The filter that will be used to query channels.
+     */
+    private fun createQueryChannelsFilter(filter: FilterObject, searchQuery: String): FilterObject {
```
Comment (filbabic) [inline stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/channel/ChannelListViewModel.kt:126]: This is really nice! Well done, I'm glad we can support this and optimize it to some extent (with the notExists check).

I've tested it and it works correctly, so all seems to be there! :]

**Review by filbabic — APPROVED** [review]
Left just one question around the git diff on this PR. IT seems like we're pushing some stuff that should already be merged into dev 🤔

File: CHANGELOG.md (hunk 1 above)
Comment (zsmb13) [reply CHANGELOG.md:94]: It was moved from ~10 lines up where it wasn't in any section under Compose into the "Added" section

Comment (filbabic) [reply CHANGELOG.md:94]: Ohh okay yeah makes sense - I'm blind 👁️

**Comment by filbabic** [comment]
@bychkovdmitry we can leave this one off for merging if you want, until we release the latest version. But we can also merge it and have it in the current release. It's up to you!

**Pull request merged**